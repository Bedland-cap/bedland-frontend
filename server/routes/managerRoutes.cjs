/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable import/no-extraneous-dependencies */
const router = require('express').Router();
const managerController = require('../controller/managerController.cjs');

router.post('/', managerController.login);

module.exports = router;