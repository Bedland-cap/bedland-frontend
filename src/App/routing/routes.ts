import { Getters } from 'types/shared.types';

const routeFeature = [
  'homeForLoggedIn',
  'homeForNotLoggedIn',
  'homeForNotLoggedInManager',
  'payments',
  'voting',
  'messages',
  'wall',
  'buildings',
  'flats',
  'reports',
  'residents',
  'resetPassword',
] as const;
const routesUrl = [
  '/dashboard',
  '/login',
  '/login-manager',
  '/payments',
  '/voting',
  '/messages',
  '/wall',
  '/buildings',
  '/flats',
  '/reports',
  '/residents',
  '/resetPassword',
] as const;
type RouteFeature = (typeof routeFeature)[number];
export type RoutesUrl = (typeof routesUrl)[number];

type RoutesData = Getters<RouteFeature, RoutesUrl>;

const routes: RoutesData = {
  homeForLoggedIn: '/dashboard',
  homeForNotLoggedIn: '/login',
  homeForNotLoggedInManager: '/login-manager',
  payments: '/payments',
  voting: '/voting',
  messages: '/messages',
  wall: '/wall',
  buildings: '/buildings',
  flats: '/flats',
  reports: '/reports',
  residents: '/residents',
  resetPassword: '/resetPassword',
} as const;

export default routes;
