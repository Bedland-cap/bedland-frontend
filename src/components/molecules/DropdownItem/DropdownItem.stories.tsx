import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import { Provider } from 'react-redux';
import { reduxStore } from 'store/store';
import DropdownItem from './DropdownItem';

const meta = {
  title: 'Molecules/DropdownItem',
  component: DropdownItem,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <BrowserRouter>
        <ThemeProvider>
          <Provider store={reduxStore}>
            <GlobalStyle />
            {Story()}
          </Provider>
        </ThemeProvider>
      </BrowserRouter>
    ),
  ],
  parameters: {
    backgrounds: { default: 'light' },
  },
} satisfies Meta<typeof DropdownItem>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Main: Story = {
  args: {
    linkRoute: '/account',
    iconName: 'user',
    label: 'My Account',
    color: 'inputGrey',
  },
};
