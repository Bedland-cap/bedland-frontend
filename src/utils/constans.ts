export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
