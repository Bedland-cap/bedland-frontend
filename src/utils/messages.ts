const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
