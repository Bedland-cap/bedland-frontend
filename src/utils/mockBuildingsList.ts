import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
