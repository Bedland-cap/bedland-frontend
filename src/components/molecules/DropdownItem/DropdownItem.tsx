import Icon from 'components/atoms/Icon/Icon';
import Typography from 'components/atoms/Typography/Typography';
import * as Styled from './DropdownItem.styled';
import DropdownItemProps from './DropdownItem.types';

const DropdownItem = ({ linkRoute, iconName, label, color, onClick }: DropdownItemProps) => (
  <Styled.DropdownItem to={linkRoute} onClick={onClick} data-testid='dropdown-item'>
    <Icon name={iconName} size={16} color={color} isActive={false} activeColor='inputGrey' />
    <Typography color={color}>{label}</Typography>
  </Styled.DropdownItem>
);

export default DropdownItem;
