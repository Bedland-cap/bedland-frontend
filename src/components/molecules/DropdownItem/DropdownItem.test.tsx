import routes from 'App/routing/routes';
import { renderWithProviders } from 'utils/test.utils';
import { fireEvent, screen } from '@testing-library/react';

import DropdownItem from './DropdownItem';

describe('DropdownItem', () => {
  it('renders correctly', () => {
    const snapshot = renderWithProviders(
      <DropdownItem linkRoute={routes.account} iconName='user' label='My Account' color='text' />,
      {},
    );
    expect(snapshot).toMatchSnapshot();
  });
  it('dropdown item redirects correctly', async () => {
    renderWithProviders(
      <DropdownItem linkRoute={routes.account} iconName='user' label='My Account' color='text' />,
      {},
    );
    const dropdownitem = screen.getByTestId('dropdown-item');
    await fireEvent.click(dropdownitem);

    expect(window.location.href).toEqual(`http://localhost${routes.account}`);
  });
});
