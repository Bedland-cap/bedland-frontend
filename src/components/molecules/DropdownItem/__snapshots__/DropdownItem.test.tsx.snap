// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.25rem;
}

<body>
    <div>
      <a
        class="c0"
        data-testid="dropdown-item"
        href="/account"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 20 20"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-user"
          />
          <path
            d="M10.0003 10.8333C12.3015 10.8333 14.167 8.96785 14.167 6.66667C14.167 4.36548 12.3015 2.5 10.0003 2.5C7.69916 2.5 5.83368 4.36548 5.83368 6.66667C5.83368 8.96785 7.69916 10.8333 10.0003 10.8333Z"
            stroke="#1B1A1A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.67"
          />
          <path
            d="M16.667 17.5002C16.667 15.7321 15.9646 14.0364 14.7144 12.7861C13.4641 11.5359 11.7685 10.8335 10.0003 10.8335C8.23224 10.8335 6.53654 11.5359 5.2863 12.7861C4.03606 14.0364 3.33368 15.7321 3.33368 17.5002"
            stroke="#1B1A1A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.67"
          />
        </svg>
        <p
          style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(27, 26, 26);"
        >
          My Account
        </p>
      </a>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.25rem;
}

<div>
    <a
      class="c0"
      data-testid="dropdown-item"
      href="/account"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 20 20"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <mask
          id="mask-user"
        />
        <path
          d="M10.0003 10.8333C12.3015 10.8333 14.167 8.96785 14.167 6.66667C14.167 4.36548 12.3015 2.5 10.0003 2.5C7.69916 2.5 5.83368 4.36548 5.83368 6.66667C5.83368 8.96785 7.69916 10.8333 10.0003 10.8333Z"
          stroke="#1B1A1A"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.67"
        />
        <path
          d="M16.667 17.5002C16.667 15.7321 15.9646 14.0364 14.7144 12.7861C13.4641 11.5359 11.7685 10.8335 10.0003 10.8335C8.23224 10.8335 6.53654 11.5359 5.2863 12.7861C4.03606 14.0364 3.33368 15.7321 3.33368 17.5002"
          stroke="#1B1A1A"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.67"
        />
      </svg>
      <p
        style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(27, 26, 26);"
      >
        My Account
      </p>
    </a>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
